src/connectors/query-rules/__tests__/connectQueryRules-test.ts

```diff
@@ -252,6 +252,51 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/query-rules
         expect(brandFilterSpy).toHaveBeenCalledWith(['Samsung', 'Apple']);
       });
 
+      test('adds ruleContexts to search parameters from numeric refinements', () => {
+        const priceFilterSpy = jest.fn(() => [500]);
+        const helper = createFakeHelper();
+        const widget = makeWidget({
+          trackedFilters: {
+            price: priceFilterSpy,
+          },
+        });
+
+        widget.init({ helper, state: helper.state, instantSearchInstance: {} });
+
+        expect(helper.getQueryParameter('ruleContexts')).toEqual([]);
+        expect(priceFilterSpy).toHaveBeenCalledTimes(1);
+        expect(priceFilterSpy).toHaveBeenCalledWith([]);
+
+        widget.render({
+          helper,
+          results: new SearchResults(helper.getState(), [{}, {}]),
+          instantSearchInstance: {},
+        });
+
+        expect(helper.getQueryParameter('ruleContexts')).toEqual([]);
+        expect(priceFilterSpy).toHaveBeenCalledTimes(2);
+        expect(priceFilterSpy).toHaveBeenCalledWith([]);
+
+        helper.setQueryParameter('numericRefinements', {
+          price: {
+            '<=': [500, 400],
+            '>=': [100],
+          },
+        });
+
+        widget.render({
+          helper,
+          results: new SearchResults(helper.getState(), [{}, {}]),
+          instantSearchInstance: {},
+        });
+
+        expect(helper.getQueryParameter('ruleContexts')).toEqual([
+          'ais-price-500',
```

Comment:
We only send one rule context? Is it expected?

Reply:
Yes, the `trackedFilters` value only returns `500` in the test:

```js
const widget = makeWidget({
  trackedFilters: {
    price: () => [500],
  },
});
```

See https://github.com/algolia/instantsearch.js/pull/3597/files#diff-1586d17bb858d5402e71bd206ba2d78fR256

Reply:
Ok but it doesn't seems that we test the behaviour where we return all the value from a refinement. We only test the behaviour with hard coded value.

Reply:
We test that the function is called we all the values for this: https://github.com/algolia/instantsearch.js/pull/3597/files#diff-1586d17bb858d5402e71bd206ba2d78fR297.

I will add a test that checks that all the values can effectively be tracked.